client/client_update_test.go

```diff
@@ -26,35 +26,97 @@ import (
 )
 
 const correctUpdateResponse = `{
-"image": {
-"uri": "https://menderupdate.com",
-"checksum": "checksum",
-"name": "core-image-base"
-},
-"id": "13876-123132-321123"
+	"id": "deplyoment-123",
+	"artifact": {
+		"source": {
+			"uri": "https://menderupdate.com",
+			"expire": "2016-03-11T13:03:17.063+0000"
+		},
+		"device_types_compatible": ["BBB"],
+		"artifact_name": "myapp-release-z-build-123"
+	}
+}`
+
+const correctUpdateResponseMultipleDevices = `{
+	"id": "deplyoment-123",
+	"artifact": {
+		"source": {
+			"uri": "https://menderupdate.com",
+			"expire": "2016-03-11T13:03:17.063+0000"
+		},
+		"device_types_compatible": [
+			"BBB",
+			"ELC AMX",
+			"IS 3"
+		],
+		"artifact_name": "myapp-release-z-build-123"
+	}
+}`
+
+const updateResponseEmptyDevices = `{
+	"id": "deplyoment-123",
+	"artifact": {
+		"source": {
+			"uri": "https://menderupdate.com",
+			"expire": "2016-03-11T13:03:17.063+0000"
+		},
+		"device_types_compatible": [],
+		"artifact_name": "myapp-release-z-build-123"
+	}
 }`
 
 const malformedUpdateResponse = `{
-"image": {
-"non_existing": "https://menderupdate.com",
-"checksum": "Hello, world!",
-},
-"bad_field": "13876-123132-321123"
+	"id": "deplyoment-123",
+	"bad_field": "13876-123132-321123"
```

Comment:
I think you need a comma at the end here. I think the test doesn't catch this because it's looking for failure, but it's failing for the wrong reason (parse error instead of missing field).